orchestra/src/lib.rs

```diff
@@ -496,6 +496,12 @@ where
 	/// Send a direct message to some other `Subsystem`, routed based on message type.
 	async fn send_message(&mut self, msg: OutgoingMessage);
 
+	/// Tries to send a direct message to some other `Subsystem`, routed based on message type.
+	/// This method is useful for cases where the message queue is bounded and the message is ok
+	/// to be dropped if the queue is full. If the queue is full, this method will return an error.
+	/// This method is not async and will not block the current task.
+	fn try_send_message(&mut self, msg: OutgoingMessage) -> Result<(), metered::TrySendError<()>>;
```

Comment:
```suggestion
	fn try_send_message(&mut self, msg: OutgoingMessage) -> Result<(), metered::TrySendError<OutgoingMessage>>;
```

I think returning the given message type is a sane thing to do, such that it could be re-pooled / re-tried.

Reply:
Yes, but reconstructing it requires quite a lot of boilerplate code. Anyway, I also think that we should invest some efforts to do it in a clearer way.